### DISCUSSION TITLE
This is a collection of ideas ot solve certain problems in the baseline classifiers.

### DISCUSSION THREAD

**Discussion by ChristopherMarais** [ChristopherMarais/IBBI#71] [Ideas]
Things that need to be improved are: 

* **OOD**

1. Virtual Outlier Synthesis
2. Energy-based methods

* Class-Imbalance

1. Logit Adjustment (start with this first then try Logit-Adjusted Focal Loss)
2. Focal loss (or Logit-Adjusted Focal Loss)
3. Test time adaptation
4. Hyperbolic Neural Networks
5. Retrieval-Augmented Classification

* Hierarchical classification

1. Multi-task learning
7. Local Classifier per Parent Node (LCPN)
8. Hierarchical Loss Function

* **Data efficiency**

1. Classic Data augmentation
9. Guided data augmentation by segmentation and background manipulation using generative methods.
10. Self-supervised learning (JEPA)
11. Semi-supervised learning
12. Curriculum learning
13. Transfer learning (LoRA)
14. Distillation

* **Human interpretability**

1. XAI - SHAP
4. Global XAI for concept discovery

* **Reliability**

1. Energy-based methods (for reliability on confidence measures)
6. Inference time augmentation and aggregation of classes.

* **Inference speed**

1. Max/Mojo for inference
2. Distillation of model to smaller version. 

OTHER ideas that might be worth exploring: 
* Continuous thought machines
* Semi-supervised concept induction
* Titans as an alternative to transformers
* Baby Dragon Hatchling architecture
* Train VLM instead of classifier.

**Comment by ChristopherMarais** [comment]
We need a benchmark dataset to test all of these things.

**Comment by ChristopherMarais** [comment]
Use SALT as a version of distillation to learn from other models.
